src/dotnet/ToolPackage/IToolPackageStore.cs

```diff
@@ -4,13 +4,22 @@
 using System;
 using System.Collections.Generic;
 using Microsoft.Extensions.EnvironmentAbstractions;
+using NuGet.Packaging.Core;
 
 namespace Microsoft.DotNet.ToolPackage
 {
     internal interface IToolPackageStore
     {
         DirectoryPath Root { get; }
 
-        IEnumerable<IToolPackage> GetInstalledPackages(string packageId = null);
+        DirectoryPath GetRandomStagingDirectory();
+
+        PackageIdentity GetStagedPackageIdentity(DirectoryPath stagingDirectory, string packageId);
+
+        DirectoryPath GetRootPackageDirectory(string packageId);
```

Comment:
Ideally packageId should not be string anymore anywhere in the code. But in some private method this is fine, However, it should be strong PackageIdentity on the interfaces at least

Reply:
Is nuget's identity is a string package id and a complex version?

https://github.com/NuGet/NuGet.Client/blob/1a1c474aab5c1e851b416c583ca4c6d727bd94e3/src/NuGet.Core/NuGet.Packaging.Core/PackageIdentity.cs

this is not suitable for us. As for now, CLI don't have concrete version most of the time. Make it absent is wrong since it is not a ID.

The problem we have is mixed casing. And this won't help us keep mixed casing out of the code

Reply:
Maybe keep nuget version type is useful in the future.

Reply:
I suggest not to use `PackageIdentity`, since CLI is not nuget, we don't know the package version. And have a wrapper type PackageId and just make sure it is lowercase in and out. And let it have proper equality

Reply:
I would say the change to not to use PackageIdentity don't have to be in this PR. So far it is ok

Reply:
I'm fine with backing out the `PackageIdentity` change in favor of our own type.  Let me amend.